pkg/metrics/gloo.go

```diff
@@ -0,0 +1,72 @@
+package metrics
+
+import (
+	"time"
+)
+
+//envoy_cluster_name="test-podinfo-primary-9898_gloo-system"
+
+var glooQueries = map[string]string{
+	"request-success-rate": `
+	sum(
+		rate(
+			envoy_cluster_upstream_rq{
+				envoy_cluster_name=~"{{ .Namespace }}-{{ .Name }}-[0-9a-zA-Z-]+_[0-9a-zA-Z-]+",
```

Comment:
Flagger should test the canary instance not the primary, please change the regex everywhere to:
```
{{ .Namespace }}-{{ .Name }}-canary-[0-9a-zA-Z-]+_[0-9a-zA-Z-]+
```